src/Components/Artsy.tsx

```diff
@@ -75,13 +96,16 @@ export class ContextProvider extends React.Component<ContextProps, null>
 
     this.relayEnvironment =
       props.relayEnvironment || createEnvironment(this.currentUser)
+
+    this.reactionRouter = props.reactionRouter
```

Comment:
Need to come up with a better name. I was going to use `router` but when we use `withRouter` HOC from found it occupies that key.

One way around this would be to refactor `ContextConsumer` so that it supports renderProps as opposed to HOC injection.

Reply:
I guess I don’t really understand the problem. Do context keys within one tree have to be unique?

Reply:
…and if that’s the problem, why do we need both?

Reply:
@alloy - inside of `PreloadLink` we need access to `found`, which inserts itself as `router` inside of the context. But we also need data from far exterior such as `routes` and the `resolver` (for relay). If we set that outer context var to `router` then it will clobber props coming in due to duplicate keys. As a sort of wip i'm storing those outer values under `reactionRouter`.

Reply:
Does found make itself publicly available through context or is it meant only for internal use of the router? If the latter, then found should imo rename their variable to not be so generic. Something like `_foundRouter`.